lib/sneakers/spawner.rb

```diff
@@ -14,7 +14,11 @@ def self.spawn
       worker_config = YAML.load(File.read(worker_group_config_file))
       worker_config.keys.each do |group_name|
         @pids << fork do
-          @exec_hash = {"WORKERS"=> worker_config[group_name]['classes'], "WORKER_COUNT" => worker_config[group_name]["workers"].to_s}
+          @exec_hash = {
+            "WORKERS"=> worker_config[group_name]['classes'],
+            "WORKER_COUNT" => worker_config[group_name]["workers"].to_s,
+            "PID_PATH" => worker_config[group_name]['pid_path'].to_s
```

Comment:
It is not easy to write a test based on the current code since all config parsing is embedded in the fork block. I'm considering to introduce a WorkerGroupConfig to handle the parsing, something like this:

<pre>
worker_config = new WorkerGroupConfiguration.New(YAML.load(File.read(worker_group_config_file)))
worker_config.keys.each do |group_name|
    @pids << fork do
        @exec_hash = config.env_vars(group_name)
        Kernel.exec(@exec_hash, @exec_string)
    end
</pre>